llvm/test/Transforms/LoopVectorize/vplan-printing.ll

```diff
@@ -655,10 +655,10 @@ define void @print_expand_scev(i64 %y, ptr %ptr) {
 ; CHECK-NEXT:    EMIT vp<[[CAN_IV:%.+]]> = CANONICAL-INDUCTION ir<0>, vp<[[CAN_IV_NEXT:%.+]]>
 ; CHECK-NEXT:     WIDEN-INDUCTION\l" +
 ; CHECK-NEXT:     "  %iv = phi %iv.next, 0\l" +
-; CHECK-NEXT:     "  ir<%v2>, vp<[[EXP_SCEV]]>, vp<[[VF]]>
+; CHECK-NEXT:     "  %v2 = trunc %iv, vp<[[EXP_SCEV]]>, vp<[[VF]]>
 ; CHECK-NEXT:     vp<[[DERIVED_IV:%.+]]> = DERIVED-IV ir<0> + vp<[[CAN_IV]]> * vp<[[EXP_SCEV]]>
 ; CHECK-NEXT:     vp<[[STEPS:%.+]]> = SCALAR-STEPS vp<[[DERIVED_IV]]>, vp<[[EXP_SCEV]]>
-; CHECK-NEXT:     WIDEN ir<%v3> = add nuw ir<%v2>, ir<1>
+; CHECK-NEXT:     WIDEN ir<%v3> = add nuw ir<%iv>, ir<1>
```

Comment:
How/is this related? It originally adds %v2.

Reply:
The patch changes `VPWidenIntOrFpInductionRecipe` to always use the phi node as underlying value (for consistency and to make it easier to retrieve it generically), whereas before the Trunc would be the underlying value.

This wasn't really obvious from the old printing for VPWidenIntOrFpInductionRecipe, which didn't show the result VPValue and VPValue operands properly. Modernized printing in 43045051d4114f2490bf0e6b01a7969d5c27ee04, hopefully clearer now and more in line with printing for other recipes.

Reply:
Yeah, the switch to hold Trunc rather than IV separately is clear. Could Trunc continue to be printed as the result %ir<%v2> of the widened induction even if its no longer the underlying?

Reply:
Unfortunately I don't think this would be possible, as at other places we usually print the underlying value of the VPValue, and trying to retrieve the Trunc would add a number of complexities.